src/renderer/selfkey-id/main/components/associate-did.jsx

```diff
@@ -0,0 +1,254 @@
+				}
+				this.setState({ searching: false });
+			}
+		}
+	}
+
+	resetErrors = () => {
+		this.props.dispatch(walletOperations.resetAssociateDID());
+	};
+
+	handleBackClick = evt => {
+		evt && evt.preventDefault();
+		this.props.dispatch(push('/main/selfkeyId'));
+	};
+
+	handleFieldChange = async event => {
+		let value = event.target.value;
+		this.setState({ did: value });
+	};
+
+	associateDID = async () => {
+		await this.resetErrors();
+		let did = this.state.did;
+		if (did !== '') {
+			await this.props.dispatch(walletOperations.updateWalletDID(this.props.wallet.id, did));
+		} else {
+			this.setState({ searching: false });
+		}
+	};
+
+	handleSubmit = () => {
+		this.setState({ searching: true }, async () => {
+			await this.associateDID();
+		});
+	};
+
+	render() {
+		const { classes, associateError } = this.props;
+		const { did, searching } = this.state;
+		const isEmpty = did === '' || did === undefined;
+		const hasAssociateError = associateError !== '' && associateError !== 'none' && did !== '';
+		const didInputClass = `${classes.input} ${
+			hasAssociateError && !searching ? classes.errorColor : ''
+		}`;
+
+		return (
+			<Grid
+				container
+				direction="column"
+				justify="flex-start"
+				alignItems="center"
+				spacing={32}
+			>
+				<ModalWrap className={classes.modalPosition}>
+					<Paper>
+						<CloseButtonIcon
+							onClick={this.handleBackClick}
+							className={classes.closeIcon}
+						/>
+						<ModalHeader>
+							<Grid
+								container
+								direction="row"
+								justify="flex-start"
+								alignItems="flex-start"
+							>
+								<Grid item>
+									<Typography variant="body1">
+										Associate DID with this wallet
+									</Typography>
+								</Grid>
+							</Grid>
+						</ModalHeader>
+						<ModalBody>
+							<Grid
+								container
+								direction="row"
+								justify="flex-start"
+								alignItems="flex-start"
+							>
+								<Grid item xs={2}>
+									<MergeIcon className={classes.icon} />
+								</Grid>
+								<Grid item xs={10}>
+									<Typography variant="body1" gutterBottom>
+										If you already registered on the SelfKey Network, you can
+										associate your existing DID number with this wallet. Just
+										copy/paste it bellow.
+									</Typography>
+									<br />
+									<Typography
+										variant="overline"
+										gutterBottom
+										className={classes.label}
+									>
+										DID Number
+										{searching && (
+											<React.Fragment>
+												<span className={classes.loading}>
+													<CircularProgress size={20} />
+												</span>
+												<span id="searching" className={classes.searching}>
+													Please wait. Checking the blockchain for DID{' '}
+													information.
+												</span>
+											</React.Fragment>
+										)}
+									</Typography>
+									<Input
+										placeholder="did:selfkey:"
+										name="did"
+										value={did}
```

Comment:
@eduardosmoura could you also set the error attribute of the Input field so it will highlight the field. Cheers

Reply:
@rodrigopavezi In fact, the "className={didInputClass}" attribute will do that highlight. 

const didInputClass = `${classes.input} ${hasAssociateError && !searching ? classes.errorColor:''}`;